diag_manager/fms_diag_file_object.F90

```diff
@@ -812,18 +844,57 @@ subroutine open_diag_file(this, time_step, file_is_opened)
     end select
   end select
 
-  if (diag_file%has_file_new_file_freq()) then
-    diag_file%next_open = diag_time_inc(diag_file%next_open, diag_file%get_file_new_file_freq(), &
-                                        diag_file%get_file_new_file_freq_units())
-  else
-    diag_file%next_open = diag_time_inc(diag_file%next_open, VERY_LARGE_FILE_FREQ, DIAG_DAYS)
-  endif
-
   file_is_opened = .true.
+  diag_file%is_file_open = file_is_opened
   domain => null()
   diag_file => null()
 end subroutine open_diag_file
 
+!< @brief Writes the time average variables (*_T1, *_T2, *_DT) in the netcdf file
+subroutine write_avg_time_metadata(fileobj, variable_name, dimensions, long_name, units)
+  class(FmsNetcdfFile_t), intent(inout) :: fileobj        !< The file object to write into
+  character(len=*)      , intent(in)    :: variable_name  !< The name of the time variables
+  character(len=*)      , intent(in)    :: dimensions(:)  !< The dimensions of the variable
+  character(len=*)      , intent(in)    :: long_name      !< The long_name of the variable
+  character(len=*)      , intent(in)    :: units          !< The units of the variable
+
+  !TODO harcodded double
```

Comment:
Is this for reproducibility?

Reply:
I hardcoded the double to make things easy for me.

I **think** in the old diag manager if you compile with r8 all of the time related variables were r8 and if you compile with r4 all of the time related variables were r4.